src/librustc_mir/dataflow/move_paths/mod.rs

```diff
@@ -72,6 +72,40 @@ impl<'tcx> MovePath<'tcx> {
 
         parents
     }
+
+    /// Looks for the first child of `self` for which `f` returns `true`.
```

Comment:
I often read the term "child" as meaning "immediate child", as opposed to recursively traversing the family tree.

 1. Would "descendant" be better?
 2. Regardless of whether you change the terminology, you should state in the documentation the search strategy. I.e., If you say "Looks for the first descendant (under breadth-first search) for which ...", then I think it will be 100% clear that this is not restricting its search to the immediate children (because one would not distinguish DFS/BFS for a traversal of the immediate children).